test/testfrmwk.hpp

```diff
@@ -56,6 +56,20 @@ inline bool check(const std::string& testname, bool testcond)
   }
 }
 
+// In the comparisons below, it is possible that T and U are signed and unsigned integer types, which generates warnings in some compilers.
+#if defined(_MSC_VER)
+# pragma warning(push)
+# pragma warning(disable: 4389)
+#elif defined(__clang__) && defined(__has_warning)
+# if __has_warning("-Wsign-compare")
+#  pragma clang diagnostic push
+#  pragma clang diagnostic ignored "-Wsign-compare"
+# endif
+#elif defined(__GNUC__) && !(defined(__INTEL_COMPILER) || defined(__ICL) || defined(__ICC) || defined(__ECC)) && (__GNUC__ * 100 + __GNUC_MINOR__) >= 406
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Wsign-compare"
+#endif
```

Comment:
Can't we just include config.hpp and then use BOOST_MSVC, BOOST_CLANG, and BOOST_GCC instead of _MSC_VER, __clang__, and the whole gcc low-level check you have in here ? Why should we reproduce what is already in Boost config and make the code more unreadable at the same time ?